subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -923,6 +924,20 @@ static int tls_mbedtls_handshake(struct tls_context *context, bool block)
 				continue;
 			}
 
+			ret = -EAGAIN;
+			break;
+		} else if (ret == MBEDTLS_ERR_SSL_ASYNC_IN_PROGRESS) {
+			if (block) {
+				continue;
+			}
+
+			ret = -EAGAIN;
+			break;
+		} else if (ret == MBEDTLS_ERR_SSL_CRYPTO_IN_PROGRESS) {
```

Comment:
Since the processing is the same anyway, you could just "or" the additional errors in the initial case to simplify the code.

Reply:
Sounds good to me, was torn between doing this or adding this explicitly 😄